drivers/ethernet/Kconfig.dsa

```
@@ -4,20 +4,26 @@
 #               Lukasz Majewski <lukma@denx.de>
 # SPDX-License-Identifier: Apache-2.0
 
-config ETH_DSA_SUPPORT
+config ETH_DSA_SUPPORT_LEGACY
 	bool
 	help
-	  Set by an ethernet driver that supports DSA.
+	  Set by an ethernet driver that supports DSA. This is obsolete,
+	  and only used for legacy dsa device.
 
 menuconfig NET_DSA
 	bool "Distributed Switch Architecture support"
-	depends on ETH_DSA_SUPPORT
 	help
 	  Enable Distributed Switch Architecture support. For now it
 	  only supports Kinetics and STM32 ENET drivers.
 
 if NET_DSA
 
+config NET_DSA_LEGACY
+	bool "Distributed Switch Architecture support for legacy device"
+	depends on ETH_DSA_SUPPORT_LEGACY
```

Probably want to select the `DEPRECATED` option.

```suggestion
config NET_DSA_LEGACY
	bool "Distributed Switch Architecture support for legacy device"
	select DEPRECATED
	depends on ETH_DSA_SUPPORT_LEGACY
```